### PR TITLE
adding a sendgrid mail client

### PR DESCRIPTION
**Pull request by hossamabdullah**
Fixing
https://github.com/Jumpscale/lib9/issues/92
by adding support for sendgrid

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -87,3 +87,4 @@ ENV/
 .ropeproject
 
 .DS_Store
+sendgrid.env
```
Comment (xmonader) [inline .gitignore]: why?

File: JumpScale9Lib/clients/mail/sendgrid/SendGridClient.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from sendgrid.helpers.mail import Email, Content, Mail
+
+
+
+class SendGridClient:
+
+    def __init__(self):
+        self.__jslocation__ = "j.clients.sendgrid"
+    
+    def send(self, sender, recipient, subject, message_type, message):
```
Comment (xmonader) [inline JumpScale9Lib/clients/mail/sendgrid/SendGridClient.py]: please set default value to message_type to be "text/plain")

File: JumpScale9Lib/clients/mail/sendgrid/SendGridClient.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        key = os.environ.get('SENDGRID_API_KEY', None)
+        if key is None:
+            raise KeyError('Make sure to export SENDGRID_API_KEY')
+        sg = sendgrid.SendGridAPIClient(apikey=key)
+        from_email = Email(sender)
+        to_email = Email(recipient)
+        content = Content(message_type, message)
+        mail = Mail(from_email, subject, to_email, content)
+        response = sg.client.mail.send.post(request_body=mail.get())
+        print(response.status_code)
```
Comment (xmonader) [inline JumpScale9Lib/clients/mail/sendgrid/SendGridClient.py]: lets remove prints and make sure to return status_code and response body instead

File: JumpScale9Lib/clients/mail/sendgrid/SendGridClient.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        @param recipient: the recipient of the email
+        @type recipient: string
+        @param subject: subject of the email
+        @type subject: string
+        @param message_type: mime type of the email content
+        @type message_type:  string 
+        @param message: content of the email
+        @type message: string
+        """
+        key = os.environ.get('SENDGRID_API_KEY', None)
```
Comment (xmonader) [inline JumpScale9Lib/clients/mail/sendgrid/SendGridClient.py]: Can we try to allow it to check in j.core.state.config as well?

File: JumpScale9Lib/clients/mail/sendgrid/SendGridClient.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        @param message: content of the email
+        @type message: string
+        @param message_type: mime type of the email content
+        @type message_type:  string
+        @param api_key: the api key of sendgrid
+        @type api_key: string
+        """
+        if api_key is None:
+            api_key = os.environ.get('SENDGRID_API_KEY', None)
+        if api_key is None:
```
Comment (rkhamis) [inline JumpScale9Lib/clients/mail/sendgrid/SendGridClient.py]: perhaps you can check if the SENDGRID_API_KEY is set in env, get and if not directly raise `RuntimeError`

File: docs/clients/SendGridClient.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#SendGrodClient
```
Comment (rkhamis) [inline docs/clients/SendGridClient.md]: typo

+ Perhaps make it more human friendly :)

Send Grid Client for instance

**Pull request merged**